custom_sticker_layer/src/main/java/com/photoeditorsdk/android/app/KEditorDemoActivity.kt

```diff
@@ -0,0 +1,176 @@
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        val openGallery = findViewById<Button>(R.id.openGallery)
+
+        openGallery.setOnClickListener {
+            openSystemGalleryToSelectAnImage()
+        }
+    }
+
+    fun openSystemGalleryToSelectAnImage() {
+        val intent = Intent(Intent.ACTION_PICK)
+        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.KITKAT) {
+            intent.type = "image/*"
+        } else {
+            intent.setDataAndType(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, "image/*")
+        }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R || intent.resolveActivity(packageManager) != null) {
+            startActivityForResult(intent, GALLERY_RESULT)
+        } else {
+            Toast.makeText(
+                this,
+                "No Gallery APP installed",
+                Toast.LENGTH_LONG
+            ).show()
+        }
+    }
+
+    fun openEditor(inputImage: Uri?) {
+        val settingsList = createPesdkSettingsList()
+
+        settingsList.configure<LoadSettings> {
+            it.source = inputImage
+        }
+
+        PhotoEditorBuilder(this)
+          .setSettingsList(settingsList)
+          .startActivityForResult(this, PESDK_RESULT)
+    }
+
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, intent: Intent?) {
+        super.onActivityResult(requestCode, resultCode, intent)
+
+        intent ?: return
+
+        if (resultCode == RESULT_OK && requestCode == GALLERY_RESULT) {
+            // Open Editor with some uri in this case with an image selected from the system gallery.
+            val selectedImage = intent.data
+            if (selectedImage != null) {
+                openEditor(selectedImage)
+            }
+        } else if (resultCode == RESULT_OK && requestCode == PESDK_RESULT) {
+            // Editor has saved an Image.
+            val data = EditorSDKResult(intent)
+
+            Log.i("PESDK", "Source image is located here ${data.sourceUri}")
+            Log.i("PESDK", "Result image is located here ${data.resultUri}")
+
+            // TODO: Do something with the result image
+
+            // OPTIONAL: read the latest state to save it as a serialisation
+            val lastState = data.settingsList
+            try {
+                IMGLYFileWriter(lastState).writeJson(File(
+                    Environment.getExternalStorageDirectory(),
+                    "serialisationReadyToReadWithPESDKFileReader.json"
+                ))
+            } catch (e: IOException) {
+                e.printStackTrace()
+            }
+
+        } else if (resultCode == RESULT_CANCELED && requestCode == PESDK_RESULT) {
+            // Editor was canceled
+            val data = EditorSDKResult(intent)
+
+            val sourceURI = data.sourceUri
+            // TODO: Do something with the source...*/
+        }
+    }
+
+}
```

Comment:
could you use the source code from the latest deployed docs? That has better handling for gallery intent, doesn't use the deprecated `getExternalStorageDirectory`, and also doesn't have un-needed permission handling.